### PR TITLE
[SPARK-26820][SQL] Prints log messages when hints ignored in the analyzer

### PR DESCRIPTION
**Pull request by maropu**
## What changes were proposed in this pull request?
This pr added code to log ignored hints in `RemoveAllHints`.

## How was this patch tested?
Pass Jenkins.

### CODE REVIEW

**Comment by maropu** [comment]
@gatorsmile Does this pr matches your intention in the Jira?

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by maropu** [comment]
ping @gatorsmile @maryannxue

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -201,6 +201,19 @@ object SQLConf {
...
+  val PLAN_HINT_IGNORE_LOG_LEVEL =
+    buildConf("spark.sql.planHintIgnoreLog.level")
+      .internal()
+      .doc("Configures the log level for logging the plan hint ignored in the analyzer. " +
+        "The value can be 'trace', 'debug', 'info', 'warn', or 'error'. " +
+        "The default log level is 'trace'.")
+      .stringConf
+      .transform(_.toUpperCase(Locale.ROOT))
+      .checkValue(logLevel => Set("TRACE", "DEBUG", "INFO", "WARN", "ERROR").contains(logLevel),
+        "Invalid value for 'spark.sql.optimizer.planChangeLog.level'. Valid values are " +
```
Comment (kiszk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `spark.sql.optimizer.planChangeLog.level` -> `spark.sql.optimizer.planHintIgnoreLog.level`?

**Comment by maropu** [comment]
Retest this please

**Comment by maropu** [comment]
Retest this please

**Comment by gatorsmile** [comment]
cc @maryannxue @maropu @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -201,6 +201,19 @@ object SQLConf {
     .stringConf
     .createOptional
 
+  val PLAN_HINT_IGNORE_LOG_LEVEL =
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Does this really worth a config?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveHintsSuite.scala [hunk 3]
```diff
@@ -155,4 +160,47 @@ class ResolveHintsSuite extends AnalysisTest {
       UnresolvedHint("REPARTITION", Seq(Literal(true)), table("TaBlE")),
       Seq(errMsgRepa))
   }
+
+  private def withLogLevel(logLevel: String)(f: => Unit): String = {
```
Comment (maryannxue) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveHintsSuite.scala]: Can we reuse `SparkFunSuite.withLogAppender`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 2 above)
Comment (maryannxue) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: +1. I don't think we need to make it configurable. We can just log warning.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala [hunk 4]
```diff
@@ -134,9 +134,42 @@ object ResolveHints {
...
+      plan resolveOperatorsUp {
+        case h: UnresolvedHint =>
+          // Logs the unused plan hint before we remove it
+          ignoredHintLogger.log(h)
+          h.child
+      }
     }
-  }
 
+    private class IgnoredHintLogger {
```
Comment (maryannxue) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala]: Why do we need a new class here? Is there any reason why we can't use `HintErrorLogger`?

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala]: Yea, HintErrorLogger can covert all the case in this pr, so I'll close this. Thanks.

**Pull request closed without merging**